malloc/malloc.c

```diff
@@ -43,7 +50,19 @@ void *mem_copy(void *dest, const void *src, int n)
 */
 void *resize_memory(void *ptr, int old_size, int new_size)
 {
+    char *newptr = ptr;
+    char *new = malloc(new_size*sizeof(ptr[0]));
+    if (new_size < old_size) {
+        for (int i=0; i < new_size; i++){
+            new[i] = newptr[i];
+            if (newptr[i] == '\0'){
+                return new;
+            }
```

Comment:
As we discussed yesterday, I don't believe the `*new` string is being properly terminated.  We're not going to enter the `for` loop unless `new_size < old_size`. The condition for the loop is based off `new_size`, so the loop will stop once `i == new_size`.

When that happens, there appears to be no logic to append or change the last character of the `*new` string to be `'\0'`. Try implementing that and see if the tests run.